build Error:  obsidian-speech-to-text @ v1.0 typecheck

Bash(npm run typecheck)
  ⎿  Error: > obsidian-speech-to-text@3.0.1 typecheck
     > npx tsc -noEmit

     src/architecture/DependencyContainer.ts(188,85): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/architecture/ErrorBoundary.ts(234,84): error TS2345: Argument of type 'unknown' is not assignable to parameter
     of type 'Error | undefined'.
     src/architecture/ErrorBoundary.ts(257,53): error TS2345: Argument of type '{ message: string; stack: string |
     undefined; context: ErrorContext; severity: ErrorSeverity; }' is not assignable to parameter of type 'Error'.
       Property 'name' is missing in type '{ message: string; stack: string | undefined; context: ErrorContext; severity:
      ErrorSeverity; }' but required in type 'Error'.
     src/architecture/ErrorBoundary.ts(329,13): error TS2345: Argument of type '{ error: Error; context: ErrorContext; }'
      is not assignable to parameter of type 'Error'.
       Object literal may only specify known properties, and 'error' does not exist in type 'Error'.
     src/architecture/PluginLifecycleManager.ts(91,56): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/architecture/PluginLifecycleManager.ts(106,59): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/architecture/PluginLifecycleManager.ts(144,59): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/architecture/PluginLifecycleManager.ts(175,79): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/architecture/PluginLifecycleManager.ts(196,61): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/infrastructure/api/providers/deepgram/DeepgramAdapter.ts(106,25): error TS2345: Argument of type '{
     originalMessage: string; audioSize: number; language: string | undefined; model: string | undefined; }' is not
     assignable to parameter of type 'Error'.
       Object literal may only specify known properties, and 'originalMessage' does not exist in type 'Error'.
     src/infrastructure/api/providers/deepgram/DeepgramAdapter.ts(134,25): error TS2345: Argument of type '{
     originalMessage: string; audioSize: number; }' is not assignable to parameter of type 'Error'.
       Object literal may only specify known properties, and 'originalMessage' does not exist in type 'Error'.
     src/main-fixed.ts(46,5): error TS2564: Property 'settings' has no initializer and is not definitely assigned in the
     constructor.
     src/main-fixed.ts(222,36): error TS2339: Property 'textContent' does not exist on type 'never'.
     src/main-fixed.ts(386,27): error TS2345: Argument of type '(value: 'cursor' | 'end' | 'beginning') => Promise<void>'
      is not assignable to parameter of type '(value: string) => any'.
       Types of parameters 'value' and 'value' are incompatible.
         Type 'string' is not assignable to type '"cursor" | "end" | "beginning"'.
     src/main-refactored.ts(46,12): error TS2416: Property 'manifest' in type 'SpeechToTextPlugin' is not assignable to
     the same property in base type 'Plugin'.
       Type '{ version: string; }' is missing the following properties from type 'PluginManifest': id, name, author,
     minAppVersion, description
     src/main-refactored.ts(103,70): error TS2345: Argument of type 'this' is not assignable to parameter of type
     'Plugin'.
       Type 'SpeechToTextPlugin' is not assignable to type 'Plugin'.
         Types of property 'manifest' are incompatible.
           Type '{ version: string; }' is not assignable to type 'PluginManifest'.
     src/main-refactored.ts(140,40): error TS2345: Argument of type 'this' is not assignable to parameter of type
     'Plugin'.
     src/main-refactored.ts(262,58): error TS2345: Argument of type 'this' is not assignable to parameter of type
     'Plugin'.
     src/main-refactored.ts(274,72): error TS2345: Argument of type 'this' is not assignable to parameter of type
     'Plugin'.
     src/main-refactored.ts(388,9): error TS2554: Expected 4-5 arguments, but got 3.
     src/main.ts(359,36): error TS2339: Property 'result' does not exist on type '{ text: string; duration: number;
     autoInsert?: boolean | undefined; options?: any; }'.
     src/main.ts(361,40): error TS2339: Property 'result' does not exist on type '{ text: string; duration: number;
     autoInsert?: boolean | undefined; options?: any; }'.
     src/main.ts(362,57): error TS2339: Property 'result' does not exist on type '{ text: string; duration: number;
     autoInsert?: boolean | undefined; options?: any; }'.
     src/main.ts(370,52): error TS2339: Property 'result' does not exist on type '{ text: string; duration: number;
     autoInsert?: boolean | undefined; options?: any; }'.
     src/ui/managers/SettingsTabManager.ts(44,67): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/SettingsTabManager.ts(94,61): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/SettingsTabManager.ts(121,68): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/SettingsTabManager.ts(134,65): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/SettingsTabManager.ts(165,72): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/SettingsTabManager.ts(183,64): error TS2345: Argument of type 'unknown' is not assignable to
     parameter of type 'Error | undefined'.
     src/ui/managers/StatusBarManager.ts(55,65): error TS2345: Argument of type 'unknown' is not assignable to parameter
     of type 'Error | undefined'.
     src/ui/managers/StatusBarManager.ts(83,64): error TS2345: Argument of type 'unknown' is not assignable to parameter
     of type 'Error | undefined'.
     src/ui/managers/StatusBarManager.ts(181,61): error TS2345: Argument of type 'unknown' is not assignable to parameter
      of type 'Error | undefined'.
     src/ui/managers/StatusBarManager.ts(207,60): error TS2345: Argument of type 'unknown' is not assignable to parameter
      of type 'Error | undefined'.
     src/ui/managers/StatusBarManager.ts(267,68): error TS2345: Argument of type 'unknown' is not assignable to parameter
      of type 'Error | undefined'.

--- a/src/architecture/DependencyContainer.ts
+++ b/src/architecture/DependencyContainer.ts
@@ -185,7 +185,7 @@ export class DependencyContainer {
                     registration.instance.dispose();
                     this.logger.debug(`Disposed service: ${String(token)}`);
                 } catch (error) {
-                    this.logger.error(`Failed to dispose service ${String(token)}`, error);
+                    this.logger.error(`Failed to dispose service ${String(token)}`, error instanceof Error ? error : undefined);
                 }
             }
         });

--- a/src/architecture/ErrorBoundary.ts
+++ b/src/architecture/ErrorBoundary.ts
@@ -231,7 +231,7 @@ export class ErrorBoundary {
                     this.logger.info(`Recovered from error using strategy: ${strategy.name}`);
                     return true;
                 } catch (recoveryError) {
-                    this.logger.error(`Recovery strategy ${strategy.name} failed`, recoveryError);
+                    this.logger.error(`Recovery strategy ${strategy.name} failed`, recoveryError instanceof Error ? recoveryError : undefined);
                 }
             }
         }
@@ -254,7 +254,7 @@ export class ErrorBoundary {
         switch (severity) {
             case ErrorSeverity.CRITICAL:
             case ErrorSeverity.HIGH:
-                this.logger.error('Error occurred', errorInfo);
+                this.logger.error('Error occurred', error, errorInfo);
                 break;
             case ErrorSeverity.MEDIUM:
                 this.logger.warn('Warning', errorInfo);
@@ -325,8 +325,7 @@ export class ErrorBoundary {
      * 치명적 에러 처리
      */
     private handleCriticalError(error: Error, context: ErrorContext): void {
-        this.logger.error('CRITICAL ERROR - Too many errors detected', {
-            error,
+        this.logger.error('CRITICAL ERROR - Too many errors detected', error, {
             context
         });
 

--- a/src/architecture/PluginLifecycleManager.ts
+++ b/src/architecture/PluginLifecycleManager.ts
@@ -88,7 +88,7 @@ export class PluginLifecycleManager {
             }
 
         } catch (error) {
-            this.logger.error('Initialization failed', error);
+            this.logger.error('Initialization failed', error instanceof Error ? error : undefined);
             await this.rollback();
             throw error;
         }
@@ -103,7 +103,7 @@ export class PluginLifecycleManager {
             this.currentPhase = LifecyclePhase.READY;
             this.logger.info('Plugin initialization completed');
         } catch (error) {
-            this.logger.error('UI initialization failed', error);
+            this.logger.error('UI initialization failed', error instanceof Error ? error : undefined);
             // UI 실패는 전체 롤백하지 않고 graceful degradation
             this.handleUIInitializationError(error);
         }
@@ -141,7 +141,7 @@ export class PluginLifecycleManager {
             this.completedTasks.add(task.name);
             this.logger.debug(`Completed task: ${task.name}`);
         } catch (error) {
-            this.logger.error(`Task ${task.name} failed`, error);
+            this.logger.error(`Task ${task.name} failed`, error instanceof Error ? error : undefined);
             throw new TaskExecutionError(task.name, error);
         }
     }
@@ -172,7 +172,7 @@ export class PluginLifecycleManager {
                     await task.rollback();
                     this.logger.debug(`Rolled back task: ${task.name}`);
                 } catch (error) {
-                    this.logger.error(`Failed to rollback task ${task.name}`, error);
+                    this.logger.error(`Failed to rollback task ${task.name}`, error instanceof Error ? error : undefined);
                 }
             }
         }
@@ -193,7 +193,7 @@ export class PluginLifecycleManager {
             try {
                 await handler();
             } catch (error) {
-                this.logger.error('Cleanup handler failed', error);
+                this.logger.error('Cleanup handler failed', error instanceof Error ? error : undefined);
             }
         }
 

--- a/src/infrastructure/api/providers/deepgram/DeepgramAdapter.ts
+++ b/src/infrastructure/api/providers/deepgram/DeepgramAdapter.ts
@@ -102,7 +102,7 @@ export class DeepgramAdapter implements ITranscriber {
             if (errorObj instanceof TranscriptionError) {
                 // 빈 transcript 에러의 경우 추가 컨텍스트 제공
                 if (errorObj.code === 'EMPTY_TRANSCRIPT') {
-                    this.logger.error('DeepgramAdapter: Empty transcript - providing user guidance', {
+                    this.logger.error('DeepgramAdapter: Empty transcript - providing user guidance', errorObj, {
                         originalMessage: errorObj.message,
                         audioSize: audio.byteLength,
                         language: options?.language,
@@ -130,7 +130,7 @@ export class DeepgramAdapter implements ITranscriber {
                 
                 // 오디오 검증 에러의 경우
                 if (errorObj.code === 'INVALID_AUDIO') {
-                    this.logger.error('DeepgramAdapter: Invalid audio format', {
+                    this.logger.error('DeepgramAdapter: Invalid audio format', errorObj, {
                         originalMessage: errorObj.message,
                         audioSize: audio.byteLength
                     });

--- a/src/main-fixed.ts
+++ b/src/main-fixed.ts
@@ -43,7 +43,7 @@ const DEFAULT_SETTINGS: SpeechToTextSettings = {
 
 // 메인 플러그인 클래스
 export default class SpeechToTextPlugin extends Plugin {
-    settings: SpeechToTextSettings;
+    settings: SpeechToTextSettings = DEFAULT_SETTINGS;
     statusBarItem: HTMLElement | null = null;
     stateManager: any;
     eventManager: any;
@@ -219,7 +219,7 @@ export default class SpeechToTextPlugin extends Plugin {
                 (this.statusBarItem as any).setText(text);
             } else {
                 // Fallback: DOM API 직접 사용
-                this.statusBarItem.textContent = text;
+                (this.statusBarItem as HTMLElement).textContent = text;
             }
         }
     }
@@ -383,7 +383,7 @@ class SpeechToTextSettingTab extends PluginSettingTab {
                 .addOption('end', '문서 끝')
                 .addOption('beginning', '문서 시작')
                 .setValue(this.plugin.settings.insertPosition)
-                .onChange(async (value: 'cursor' | 'end' | 'beginning') => {
+                .onChange(async (value: string) => {
                     this.plugin.settings.insertPosition = value;
                     await this.plugin.saveSettings();
                 }));

--- a/src/main-refactored.ts
+++ b/src/main-refactored.ts
@@ -43,7 +43,6 @@ export default class SpeechToTextPlugin extends Plugin {
     
     // 설정
     public settings: any;
-    public manifest = { version: "1.0.0" };
 
     /**
      * 플러그인 로드
@@ -385,8 +384,12 @@ export default class SpeechToTextPlugin extends Plugin {
             return;
         }
 
-        new FilePickerModal(this.app, audioFiles, async (file) => {
-            await this.transcribeFile(file);
+        new FilePickerModal(this.app, {}, async (files) => {
+            for (const file of files) {
+                await this.transcribeFile(file.file);
+            }
+        }, () => {
+            // onCancel callback
         }).open();
     }
 

--- a/src/main.ts
+++ b/src/main.ts
@@ -356,18 +356,16 @@ export default class SpeechToTextPlugin extends Plugin {
         this.eventManager.on('transcription:complete', async (data) => {
             this.logger.debug('=== Transcription complete event received ===', {
                 hasData: !!data,
-                hasResult: !!data?.result,
                 hasText: !!data?.text,
-                hasResultText: !!data?.result?.text,
-                textLength: data?.text?.length || data?.result?.text?.length || 0,
+                textLength: data?.text?.length || 0,
                 autoInsert: this.settings.autoInsert
             });
             
             this.stateManager.setState({ status: 'completed' });
             new Notice('Transcription completed successfully');
             
-            // Extract text from either data.text or data.result.text
-            const textToInsert = data.text || data.result?.text;
+            // Extract text from data.text
+            const textToInsert = data.text;
             
             if (!textToInsert) {
                 this.logger.error('No text found in transcription complete event', undefined, { data });

--- a/src/ui/managers/SettingsTabManager.ts
+++ b/src/ui/managers/SettingsTabManager.ts
@@ -41,7 +41,7 @@ export class SettingsTabManager implements IDisposable {
 
             this.logger.info('SettingsTab initialized successfully');
         } catch (error) {
-            this.logger.error('Failed to initialize SettingsTab', error);
+            this.logger.error('Failed to initialize SettingsTab', error instanceof Error ? error : undefined);
             // SettingsTab 실패는 치명적이지 않으므로 에러를 던지지 않음
             this.handleInitializationError(error);
         }
@@ -91,7 +91,7 @@ export class SettingsTabManager implements IDisposable {
             this.settingsTab = settingsTab;
             this.logger.debug('SettingsTab created and registered successfully');
         } catch (error) {
-            this.logger.error('Error creating SettingsTab', error);
+            this.logger.error('Error creating SettingsTab', error instanceof Error ? error : undefined);
             this.settingsTab = null;
         }
     }
@@ -118,7 +118,7 @@ export class SettingsTabManager implements IDisposable {
 
             return tab;
         } catch (error) {
-            this.logger.error('Failed to instantiate SettingsTab', error);
+            this.logger.error('Failed to instantiate SettingsTab', error instanceof Error ? error : undefined);
             return null;
         }
     }
@@ -131,7 +131,7 @@ export class SettingsTabManager implements IDisposable {
             this.plugin.addSettingTab(settingsTab);
             this.logger.debug('SettingsTab registered with plugin');
         } catch (error) {
-            this.logger.error('Failed to register SettingsTab', error);
+            this.logger.error('Failed to register SettingsTab', error instanceof Error ? error : undefined);
             throw error;
         }
     }
@@ -162,7 +162,7 @@ export class SettingsTabManager implements IDisposable {
             this.settingsTab = fallbackTab;
             this.logger.info('Fallback SettingsTab created');
         } catch (error) {
-            this.logger.error('Failed to create fallback SettingsTab', error);
+            this.logger.error('Failed to create fallback SettingsTab', error instanceof Error ? error : undefined);
         }
     }
 
@@ -180,7 +180,7 @@ export class SettingsTabManager implements IDisposable {
                 this.logger.debug('SettingsTab refreshed');
             }
         } catch (error) {
-            this.logger.error('Failed to refresh SettingsTab', error);
+            this.logger.error('Failed to refresh SettingsTab', error instanceof Error ? error : undefined);
         }
     }
 

--- a/src/ui/managers/StatusBarManager.ts
+++ b/src/ui/managers/StatusBarManager.ts
@@ -52,7 +52,7 @@ export class StatusBarManager implements IDisposable {
 
             this.logger.info('StatusBar initialized successfully');
         } catch (error) {
-            this.logger.error('Failed to initialize StatusBar', error);
+            this.logger.error('Failed to initialize StatusBar', error instanceof Error ? error : undefined);
             // StatusBar 실패는 치명적이지 않으므로 에러를 던지지 않음
         }
     }
@@ -80,7 +80,7 @@ export class StatusBarManager implements IDisposable {
             this.statusBarItem = item;
             this.logger.debug('StatusBar item created successfully');
         } catch (error) {
-            this.logger.error('Error creating StatusBar item', error);
+            this.logger.error('Error creating StatusBar item', error instanceof Error ? error : undefined);
             this.statusBarItem = null;
         }
     }
@@ -178,7 +178,7 @@ export class StatusBarManager implements IDisposable {
             }
 
         } catch (error) {
-            this.logger.error('Failed to update StatusBar', error);
+            this.logger.error('Failed to update StatusBar', error instanceof Error ? error : undefined);
         }
     }
 
@@ -204,7 +204,7 @@ export class StatusBarManager implements IDisposable {
                 this.statusBarItem.textContent = '';
             }
         } catch (error) {
-            this.logger.error('Failed to clear StatusBar', error);
+            this.logger.error('Failed to clear StatusBar', error instanceof Error ? error : undefined);
         }
     }
 
@@ -264,7 +264,7 @@ export class StatusBarManager implements IDisposable {
                     this.statusBarItem.parentNode.removeChild(this.statusBarItem);
                 }
             } catch (error) {
-                this.logger.error('Error removing StatusBar item', error);
+                this.logger.error('Error removing StatusBar item', error instanceof Error ? error : undefined);
             }
             
             this.statusBarItem = null;

